### ISSUE TITLE
Uploaded .tar.gz but can't uncompress

### ISSUE THREAD

**Issue by mgscreativa** [prasathmani/tinyfilemanager#1002]
Downloaded tfm 2.5.3 at commit [bf024c7](https://github.com/prasathmani/tinyfilemanager/commit/bf024c7c841be7600e41cf8e0564786ffad1d7a3) then uploaded a tar.gz file an dtried to uncompress it, but no uncompress option given. See screenshot.

Does TFM supports tar.gz uncompression? Thanks!

![imagen](https://user-images.githubusercontent.com/1286632/228843601-025bd610-0f6b-4260-8d9c-0052ff002ae0.png)

**Comment by ner00** [comment]
Apparently not, could be implemented though: https://www.php.net/manual/en/function.gzdecode.php

**Comment by prasathmani** [comment]
<img width="606" alt="image" src="https://user-images.githubusercontent.com/11345598/228975190-37a8a647-1c20-4cd5-945e-0a2b4052ace3.png">

**Comment by mgscreativa** [comment]
Thanks for the replies. Regarding the requirements I have PHP 7.4 and this list of extensions enabled, can't find something like tar extension for php, but phar is enabled

- 10-mysqlnd.ini
- 10-opcache.ini
- 10-pdo.ini
- 15-xml.ini
- 20-apcu.ini
- 20-bcmath.ini
- 20-bz2.ini
- 20-calendar.ini
- 20-ctype.ini
- 20-curl.ini
- 20-dom.ini
- 20-exif.ini
- 20-ffi.ini
- 20-fileinfo.ini
- 20-ftp.ini
- 20-gd.ini
- 20-gettext.ini
- 20-iconv.ini
- 20-imagick.ini
- 20-imap.ini
- 20-intl.ini
- 20-json.ini
- 20-ldap.ini
- 20-mbstring.ini
- 20-mcrypt.ini
- 20-mysqli.ini
- 20-pdo_mysql.ini
- 20-pdo_sqlite.ini
- 20-phar.ini
- 20-posix.ini
- 20-pspell.ini
- 20-readline.ini
- 20-shmop.ini
- 20-simplexml.ini
- 20-soap.ini
- 20-sockets.ini
- 20-sqlite3.ini
- 20-sysvmsg.ini
- 20-sysvsem.ini
- 20-sysvshm.ini
- 20-tidy.ini
- 20-tokenizer.ini
- 20-xdebug.ini
- 20-xmlreader.ini
- 20-xmlrpc.ini
- 20-xmlwriter.ini
- 20-xsl.ini
- 20-zip.ini

**Comment by ner00** [comment]
@mgscreativa if you're interested, I implemented basic gzip support: https://github.com/ner00/tinyfilemanager/tree/gzip
Note that you will need zlib extension enabled, which in you case seems you need to install it first.

Archive content info is not even bare minimum, focus was primarily on being able to extract those archives.

@prasathmani I'm not creating a PR for this for 2 reasons: it's pretty bad/incomplete and, as you hinted before, raises the environment requirements for TFM.

**Comment by mgscreativa** [comment]
@ner00 Will try. Remember that .tar.gzcould be .tgz too…

**Comment by ner00** [comment]
@mgscreativa Rename it, otherwise there's no end to that conversation.

**Comment by ner00** [comment]
Added tgz as supported extension too.

**Comment by mgscreativa** [comment]
Cool! rill try with a .tar.gz of 14 gb I need to decompress...

**Comment by ner00** [comment]
Btw, I just pushed a small change to verify that `fopen()` before `fwrite()`

**Comment by mgscreativa** [comment]
Ok, tested with 1.8GB tar.gz file, the error is:

`Fatal error: Allowed memory size of 134217728 bytes exhausted (tried to allocate 4096 bytes) in /tinyfilemanager.php on line 2664`

Tried putting a .user.ini with memory_limit = 1G also with 4GB and 16 GB the error persisted

**Comment by ner00** [comment]
This means that the value you are changing is not having any effect. Which either means it's teh wrong config gile (some installs have multiple php.ini per environment), or you didn't restart the webserver (Apache, nginx, etc) after making the change. Just tried it with a 3.27 GB and worked fine after I increased `memory_limit` and restarted my webserver.

Note: It's a bit weird that your error makes it seem that you have more memory than it needs to allocate, in my case I had 512MB and the error was: `Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 536870912 bytes)` which makes sense because I tried to allocate all of it. Anyway, that might be down to the environment, but it works for me once I increase and restart the webserver.
